### ISSUE TITLE
PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434

### ISSUE THREAD

**Issue by netruner** [glpi-project/glpi#6937]
**Describe the bug**

Block web session user and must restart web browser

**Page(s) URL**

is randomize

**To reproduce**

Normal use GLPI

**Expected behavior**

**Logs**

```
PHP Fatal error:  Call to a member function addRecord() on null in [Wed Feb 12 14:11:40.337192 2020] [:error] [pid 6184] [client 10.0.0.180:61705] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/front/entity.php
[Wed Feb 12 14:19:50.070600 2020] [:error] [pid 6089] [client 10.0.10.74:51591] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/
[Wed Feb 12 14:19:59.974712 2020] [:error] [pid 6089] [client 10.0.0.180:62659] PHP Warning:  include_once(): Failed opening '' for inclusion (include_path='.:/usr/share/php:/usr/share/pear') in /var/www/glpi/front/css.php on line 43, referer: https://helpdesk.usdk.pl/front/massiveaction.php
[Wed Feb 12 14:19:59.974765 2020] [:error] [pid 6089] [client 10.0.0.180:62659] PHP Fatal error:  Class 'Html' not found in /var/www/glpi/front/css.php on line 45, referer: https://helpdesk.usdk.pl/front/massiveaction.php
[Wed Feb 12 14:25:10.224771 2020] [:error] [pid 6866] [client 10.0.10.20:32934] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/index.php?noAUTO=1
[Wed Feb 12 14:30:55.572894 2020] [:error] [pid 6821] [client 10.0.50.120:57200] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/index.php?redirect=%2Ffront%2Fhelpdesk.public.php&error=3
[Wed Feb 12 14:37:05.220106 2020] [:error] [pid 6903] [client 10.0.10.74:55646] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/plugins/mydashboard/front/menu.php
[Wed Feb 12 14:44:18.400816 2020] [:error] [pid 7059] [client 10.0.0.193:38076] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/front/ticket.php
[Wed Feb 12 14:51:19.917318 2020] [:error] [pid 7060] [client 10.0.0.180:50131] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/front/ticket.php
[Wed Feb 12 15:45:00.974347 2020] [:error] [pid 7061] [client 10.0.181.247:49398] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://awaria.usdk.pl/front/helpdesk.public.php?create_ticket=1
[Wed Feb 12 23:05:19.248947 2020] [:error] [pid 19276] [client 172.16.1.1:55252] PHP Fatal error:  Call to a member function addRecord() on null in /var/www/glpi/inc/toolbox.class.php on line 434, referer: https://helpdesk.usdk.pl/plugins/formcreator/front/form.form.php?id=23
```

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

[code]   GLPI 9.4.5 ( => /var/www/glpi) Installation mode: TARBALL

Operating system: Linux glpi 4.9.0-8-amd64 #1 SMP Debian 4.9.144-3.1 (2019-02-19) x86_64 PHP 5.6.40-0+deb8u1 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc, apcu, 	bcmath, bz2, calendar, ctype, curl, date, dba, dom, ereg, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, json, 	ldap, libxml, mbstring, mhash, mysql, mysqli, openssl, pcre, pdo_mysql, posix, readline, session, shmop, snmp, soap, sockets, 	standard, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlwriter, zip, zlib) Setup: max_execution_time="30" memory_limit="128M" post_max_size="50M" safe_mode="" session.save_handler="files" 	upload_max_filesize="51M"  Software: Apache/2.4.25 (Debian) (Apache/2.4.25 (Debian) Server at helpdesk.usdk.pl Port 443) 	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 Safari/605.1.15 Server Software: Debian 9.6 	Server Version: 10.1.37-MariaDB-0+deb9u1 	Server SQL Mode:  	Parameters: glpi@10.0.50.112/glpi 	Host info: 10.0.50.112 via TCP/IP 	 mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed xml extension is installed ldap extension is installed imap extension is installed Zend OPcache extension is installed APCu extension is installed exif extension is installed xmlrpc extension is not present CAS extension is not present Database version seems correct (10.1.37) - Perfect! /var/www/glpi/files/_log : OK /var/www/glpi/config : OK /var/www/glpi/files : OK /var/www/glpi/files/_dumps : OK /var/www/glpi/files/_sessions : OK /var/www/glpi/files/_cron : OK /var/www/glpi/files/_graphs : OK /var/www/glpi/files/_lock : OK /var/www/glpi/files/_plugins : OK /var/www/glpi/files/_tmp : OK /var/www/glpi/files/_cache : OK /var/www/glpi/files/_rss : OK /var/www/glpi/files/_uploads : OK /var/www/glpi/files/_pictures : OK Web access to files directory is protected : OK

htmLawed version 1.2.4 in (/var/www/glpi/lib/htmlawed) phpmailer/phpmailer version 6.0.7 in (/var/www/glpi/vendor/phpmailer/phpmailer/src) simplepie/simplepie version 1.5.2 in (/var/www/glpi/vendor/simplepie/simplepie/library) tecnickcom/tcpdf version 6.2.26 in (/var/www/glpi/vendor/tecnickcom/tcpdf) michelf/php-markdown in (/var/www/glpi/vendor/michelf/php-markdown/Michelf) true/punycode in (/var/www/glpi/vendor/true/punycode/src) iamcal/lib_autolink in (/var/www/glpi/vendor/iamcal/lib_autolink) sabre/vobject in (/var/www/glpi/vendor/sabre/vobject/lib) zendframework/zend-cache in (/var/www/glpi/vendor/zendframework/zend-cache/src) zendframework/zend-i18n in (/var/www/glpi/vendor/zendframework/zend-i18n/src) zendframework/zend-serializer in (/var/www/glpi/vendor/zendframework/zend-serializer/src) monolog/monolog in (/var/www/glpi/vendor/monolog/monolog/src/Monolog) sebastian/diff in (/var/www/glpi/vendor/sebastian/diff/src) elvanto/litemoji in (/var/www/glpi/vendor/elvanto/litemoji/src) symfony/console in (/var/www/glpi/vendor/symfony/console) leafo/scssphp in (/var/www/glpi/vendor/leafo/scssphp/src) paragonie/random_compat in (/var/www/glpi/vendor/paragonie/random_compat/lib)

datainjection        Name: Data injection                 Version: 2.7.1      State: Enabled 	fields               Name: Dodatkowe pola                 Version: 1.10.2     State: Enabled 	formcreator          Name: Form Creator                   Version: 2.8.0      State: Enabled 	fusioninventory      Name: FusionInventory                Version: 9.4+2.4    State: Enabled 	mod                  Name: GLPI Modifications             Version: 1.3.0      State: Enabled 	moreticket           Name: More ticket                    Version: 1.5.0      State: Enabled 	mydashboard          Name: My Dashboard                   Version: 1.7.1      State: Not activated 	ocsinventoryng       Name: OCS Inventory NG               Version: 1.6.0      State: Enabled 	printercounters      Name: Printer counters               Version: 1.6.0      State: Not activated 	dashboard            Name: Pulpit                         Version: 0.9.6      State: Enabled 	telegrambot          Name: TelegramBot                    Version: 2.0.0      State: Enabled 	genericobject        Name: Zarządzanie obiektami          Version: 2.7.0      State: Enabled 	order                Name: Zarządzanie zamówieniami       Version: 2.4.0      State: Not activated

Host: 10.0.50.112Connection: OkUse the OCSNG software dictionary: No
[/code]

**Comment by cedric-anne** [comment]
Hi,

It is hard to understand what is the root cause of this error. Can you try to deactivate plugins, with an admin account from a different browser, when it happens, and see if problem persist?

Regards

**Comment by netruner** [comment]
After updating the system, GLPI began to crash. One of the most common error was the above. We've updates to all the plugins and/or we've disabled for testing. However, GLPI hung still blocking the browser session from time to time. Only the browser restart and apache helped. Yesterday we solved the hang problem, it was a problem with IMAP checking emails. However, the error still appears, however, no longer hangs sessions and there are very few
